modules/yugabytedb/yugabytedb.go

```diff
@@ -0,0 +1,183 @@
+// [*Container.YSQLConnectionString] and [*Container.YCQLConfigureClusterConfig]
+// methods to use the container in their respective clients.
+func Run(ctx context.Context, img string, opts ...testcontainers.ContainerCustomizer) (*Container, error) {
+	req := testcontainers.ContainerRequest{
+		Image: img,
+		Cmd:   []string{"bin/yugabyted", "start", "--background=false"},
+		WaitingFor: wait.ForAll(
+			wait.ForLog("YugabyteDB Started").WithOccurrence(1),
+			wait.ForLog("Data placement constraint successfully verified").WithOccurrence(1),
+			wait.ForListeningPort(ysqlPort),
+			wait.ForListeningPort(ycqlPort),
+		),
+		ExposedPorts: []string{ycqlPort, ysqlPort},
+		Env: map[string]string{
+			ycqlKeyspaceEnv:         ycqlKeyspace,
+			ycqlUserNameEnv:         ycqlUserName,
+			ycqlPasswordEnv:         ycqlPassword,
+			ysqlDatabaseNameEnv:     ysqlDatabaseName,
+			ysqlDatabaseUserEnv:     ysqlDatabaseUser,
+			ysqlDatabasePasswordEnv: ysqlDatabasePassword,
+		},
+	}
+
+	genericContainerReq := testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	}
+
+	for _, opt := range opts {
+		if err := opt.Customize(&genericContainerReq); err != nil {
+			return nil, fmt.Errorf("customize: %w", err)
+		}
+	}
+
+	container, err := testcontainers.GenericContainer(ctx, genericContainerReq)
+	var c *Container
+	if container != nil {
+		c = &Container{
+			Container:            container,
+			ysqlDatabaseName:     req.Env[ysqlDatabaseNameEnv],
+			ysqlDatabaseUser:     req.Env[ysqlDatabaseUserEnv],
+			ysqlDatabasePassword: req.Env[ysqlDatabasePasswordEnv],
+		}
+	}
+
+	if err != nil {
+		return c, fmt.Errorf("generic container: %w", err)
+	}
+
+	return c, nil
+}
+
+// YSQLConnectionString returns a connection string for the yugabyteDB container
+// using the configured database name, user, password, port, host and additional
+// arguments.
+// Additional arguments are appended to the connection string as query parameters
+// in the form of key=value pairs separated by "&".
+func (y *Container) YSQLConnectionString(ctx context.Context, args ...string) (string, error) {
+	host, err := y.Host(ctx)
+	if err != nil {
+		return "", fmt.Errorf("host: %w", err)
+	}
+
+	mappedPort, err := y.MappedPort(ctx, ysqlPort)
+	if err != nil {
+		return "", fmt.Errorf("mapped port: %w", err)
+	}
+
+	return fmt.Sprintf(
+		"postgres://%s:%s@%s/%s?%s",
+		y.ysqlDatabaseUser,
+		y.ysqlDatabasePassword,
+		net.JoinHostPort(host, mappedPort.Port()),
+		y.ysqlDatabaseName,
+		strings.Join(args, "&"),
+	), nil
+}
+
+// YCQLConfigureClusterConfig merges the provider [gocql.ClusterConfig] with the
+// yugabyteDB container configuration.
+// The method sets the hosts, keyspace and authenticator fields of the cluster
+// configuration to the values of the yugabyteDB container.
+func (y *Container) YCQLConfigureClusterConfig(ctx context.Context, cfg *gocql.ClusterConfig) error {
```

Comment:
blocker: we usually do not bundle any client/sdk into the module, to let users choose what they need to connect to the given technology. I'd recommend having this method as a helper function in the tests/examples, so people as a reference on how to bootstrap the config from the container state.

Therefore, the `yugabyte/gocql` package would be used just in the tests, not in the "production" container.

Wdyt?

Reply:
Fair enough. I agree, this could become a maintenance hell with versioning etc.. 
I'll make the changes 👍

Reply:
✅ Attended

Reply:
@mdelapenya Regarding this point may I suggest we add something around it in the contributing guidelines ?